docs/reference/content/bson/pojos.md

```diff
@@ -196,6 +196,79 @@ all serializable because they are bound to the concrete types `Integer`, `String
 On their own, instances of `GenericTree` or `GenericClass` are not serializable by the `PojoCodec`. This is because the runtime type parameter
 information is erased by the JVM, and the type parameters cannot be specialized accurately.
 
+The 3.6 release of the driver further improves generic support with the addition of PropertyCodecProviders. The `PropertyCodecProvider` API 
+allows type safe support of container types, by providing concrete type parameters for the generic types as declared in the POJO.
+
+A great use of the `PropertyCodecProvider` API could be to add support for the Java 8 `Optional` class. The following example creates a 
+`OptionalPropertyCodecProvider`:
+
+```java
+public class OptionalPropertyCodecProvider implements PropertyCodecProvider {
```

Comment:
Use Guava's [Optional](https://google.github.io/guava/releases/19.0/api/docs/com/google/common/base/Optional.html) class as the example instead, as it's less likely that the driver would ever support that out of the box.

Reply:
Done